src/header/components/HeaderSearch.tsx

```diff
@@ -86,34 +86,37 @@ const HeaderSearch: FunctionComponent = () => {
     }
   }
 
-  const onFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    e.stopPropagation()
-
-    // If a suggestion is selected use that one, otherwise submit the form
-    if (selectedElement) {
-      document.querySelector<HTMLAnchorElement>(`.${ACTIVE_ITEM_CLASS}`)?.click()
-    } else {
-      const queryString = {
-        term: inputValue.trim(),
+  const onFormSubmit = useCallback(
+    (e: React.FormEvent<HTMLFormElement>) => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      // If a suggestion is selected use that one, otherwise submit the form
+      if (selectedElement) {
+        document.querySelector<HTMLAnchorElement>(`.${ACTIVE_ITEM_CLASS}`)?.click()
+      } else {
+        const queryString = {
+          term: inputValue.trim(),
+        }
+        const query = new URLSearchParams(
+          Object.entries(queryString).reduce(
+            (acc, [key, value]) => (value ? { ...acc, [key]: value } : acc),
+            {},
+          ),
+        )
```

Comment:
Can't we put this hash directly into the constructor?
```suggestion
        const query = new URLSearchParams({
          term: inputValue.trim(),
        })
```

Reply:
Possibly, but that code was already in the component.

Reply:
Sure, but no harm leaving things better than they were before, no?

Reply:
Very true. But making a change like that requires adding tests, which there are none for this component. I'll create a different PR for that.

Reply:
I've simplified the code and tested it locally, looks like it was just redundant.